storagedriver/s3/s3.go

```diff
@@ -580,6 +580,16 @@ func (d *Driver) Delete(path string) error {
 	return nil
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string) (string, error) {
+	if !storagedriver.PathRegexp.MatchString(path) {
+		return "", storagedriver.InvalidPathError{Path: path}
+	}
+
+	return d.Bucket.SignedURL(d.s3Path(path), time.Now().Add(24*time.Hour)), nil
```

Comment:
We can add that as an argument to `URLFor()`, but certain drivers will ignore the TTL. As long as we define the semantic properly and stick with it then it's ok.